wsmaster/che-core-api-agent/src/main/java/org/eclipse/che/api/agent/server/AgentRegistry.java

```diff
@@ -0,0 +1,90 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.agent.server;
+
+import org.eclipse.che.api.agent.server.exception.AgentException;
+import org.eclipse.che.api.agent.server.exception.AgentNotFoundException;
+import org.eclipse.che.api.agent.shared.model.Agent;
+import org.eclipse.che.api.agent.shared.model.AgentKey;
+
+import java.util.Collection;
+import java.util.Set;
+
+/**
+ * @author Anatoliy Bazko
+ */
+public interface AgentRegistry {
+
+    /**
+     * Creates {@link Agent} of the specific version.
+     *
+     * @param name
+     *      the name of the agent
+     * @param version
+     *      the version of the agent
+     * @return {@link Agent}
+     * @throws AgentNotFoundException
+     *      if agent not found in the registry
+     * @throws AgentException
+     *      if agent can't be created or other unexpected error occurred
+     */
+    Agent createAgent(String name, String version) throws AgentException;
```

Comment:
It doesn't create Agent. As far as I understand it returns agent by name and version